docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
 # Developer Guide
 
-## Design & Implementation
+### Table of Contents
+* **[1. Introduction](#1-introduction)**
+* **[2. Overview of the SHOCO application](#2-overview-of-the-shoco-application)**
+* **[3. Implementation](#3-implementation)**
+    + [3.1 Add feature](#31-add-feature)
+    + [3.2 Edit feature](#32-edit-feature)
+    + [3.3 Delete feature](#33-delete-feature)
+    + [3.4 Find feature](#34-find-feature)
+    + [3.5 Mark and Unmark feature](#35-mark-and-unmark-feature)
+    + [3.6 Display feature](#36-display-feature)
+    + [3.7 Clear list feature](#37-clear-list-feature)
+    + [3.8 Set budget feature](#38-set-budget-feature)
+    + [3.9 Reset budget feature](#39-reset-budget-feature)
+    + [3.10 View help feature](#310-view-help-feature)
+    + [3.11 Exit program feature](#311-exit-program-feature)
+ * **[Appendix A: Product Scope](#appendix-a-product-scope)**
+ * **[Appendix B: User Stories](#appendix-b-user-stories)**
+ * **[Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)**
+ * **[Appendix D: Glossary](#appendix-d-glossary)**
+ * **[Appendix E: Instructions for Manual Testing](#appendix-e-instructions-for-manual-testing)**
+ 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Introduction
 
+### Purpose of this guide
+This guide describes the software architecture and design of the SHOCO application.
+It will evolve throughout the design and implementation of each SHOCO release. 
+Currently, this document is for the first public release of the application, SHOCO v1.0.
```

Comment:
Small thing, but should this be SHOCO v2.0? I saw that the Appendix contains info about v2.0